website/addons/s3/static/s3NodeConfig.js

```diff
@@ -91,31 +91,45 @@ ViewModel.prototype.updateBucketList = function(){
 
 ViewModel.prototype.selectBucket = function() {
     var self = this;
+
+    // The current version of boto does not handle bucket names containing periods
+    var isValidBucket = /^(?!.*(\.\.|-\.))[^.][a-z0-9\d-]{2,61}[^.]$/;
+
     self.loading(true);
-    return $osf.postJSON(
+
+    if (isValidBucket.exec(self.selectedBucket()) == null) {
+        bootbox.alert({
+            title: 'Invalid bucket name',
+            message: 'Sorry, the S3 addon only supports bucket names without periods',
+            callback: function() {
+                self.loading(false);
```

Comment:
Why set `loading(false)` in the callback? Would it make sense to do immediately in this block (~L101), so that any loading indicators are hidden.